### PR TITLE
RavenDB-10732: [merged from v4.0] static smapsReader

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Sparrow/LowMemory/LowMemoryNotification.cs [hunk 1]
```diff
@@ -130,6 +131,7 @@ public static void Initialize(Size lowMemoryThreshold, float commitChargeThresho
         private readonly ManualResetEvent _simulatedLowMemory = new ManualResetEvent(false);
         private readonly ManualResetEvent _shutdownRequested = new ManualResetEvent(false);
         private readonly List<WeakReference<ILowMemoryHandler>> _inactiveHandlers = new List<WeakReference<ILowMemoryHandler>>(128);
+    private static readonly SmapsReader SmapsReader = new SmapsReader(new []{new byte[SmapsReader.BufferSize], new byte[SmapsReader.BufferSize]});
```
Comment (ayende) [inline src/Sparrow/LowMemory/LowMemoryNotification.cs]: Remove this. This is _going_ to cause us threading issues.
Instead, create this as a _local_ variable in the top function

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 2]
```diff
@@ -136,7 +136,7 @@ public Task MemorySmaps()
 
             using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
             {
-                var result = new SmapsReader().CalculateMemUsageFromSmaps<SmapsReaderJsonResults>();
+                var result = new SmapsReader(new []{new byte[SmapsReader.BufferSize], new byte[SmapsReader.BufferSize]}).CalculateMemUsageFromSmaps<SmapsReaderJsonResults>();
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: Use `ArrayPool<byte>.Shared` here, instead of allocating this

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow/LowMemory/LowMemoryNotification.cs (hunk 1 above)
Comment (aviviadi) [reply src/Sparrow/LowMemory/LowMemoryNotification.cs]: how this is better then creating the buffer inside SmapsReader ?

**Comment by ayende** [comment]
Do not edit commit messages

**Pull request closed without merging**